### PR TITLE
remove duplicated journeys because of bss

### PR DESCRIPTION
**Pull request by xlqian**
https://jira.kisio.org/browse/NAVITIAII-3447

In this PR, I replaced bss fallback by a walking fallback if the bss fallback is actually a pure walking fallback. At the end of distributed_scenario, I added another filter to remove the duplicated journeys

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py [hunk 1]
```diff
@@ -385,6 +386,18 @@ def _build_crowfly(pt_journey, entry_point, mode, places_free_access, fallback_d
...
+def _is_pure_walking(dp):
+    try:
+        if len(dp.journeys[0].sections) != 1:
+            return False
+        section = dp.journeys[0].sections[0]
+        if section.type == response_pb2.STREET_NETWORK and section.street_network.mode == response_pb2.Walking:
+            return True
+        return False
+    except IndexError:
+        return False
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py]:
```suggestion
   if len(dp.journeys > 0):
      if len(dp.journeys[0].sections) == 1:
         section =  dp.journeys[0].sections[0]
         if section.type == response_pb2.STREET_NETWORK and section.street_network.mode == response_pb2.Walking :
                        return True
   return False                        
```

File: source/jormungandr/jormungandr/scenarios/journey_filter.py [hunk 2]
```diff
@@ -647,6 +647,20 @@ def apply_final_journey_filters(response_list, instance, request):
     _filter_too_much_connections(journeys, instance, request)
 
 
+def replace_bss_tag(journeys):
+    """
+    replace the gss tag by walking, if there's no bss section in the journey
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/journey_filter.py]:
```suggestion
    replace the `bss` tag by `walking`, if there's no bss section in the journey
```

File: source/jormungandr/jormungandr/scenarios/journey_filter.py [hunk 3]
```diff
@@ -647,6 +647,20 @@ def apply_final_journey_filters(response_list, instance, request):
...
+    """
+    for j in journeys:
+        if not j.tags or "bss" not in j.tags:
+            continue
+        has_bss = any((s.type == response_pb2.BSS_RENT for s in j.sections))
+        if has_bss:
+            continue
+        j.tags.remove("bss")
+        j.tags.append("walking")
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/journey_filter.py:648]: Can it happen that there is no walking at all (e.g. the starting point is a stop point) and thus the "walking" tag should not be appended ?

Comment (xlqian) [reply source/jormungandr/jormungandr/scenarios/journey_filter.py:648]: the goal is to replace the bss tag by walking ONLY when `bss` tag is existent in `journeys.tags`. In fact, with the presence of `bss` tag, we're sure that there must be a fallback return by the `bss` connector but we're not sure if it's pure walking or bss...

Comment (pbench) [reply source/jormungandr/jormungandr/scenarios/journey_filter.py:648]: Ok, I guess the case I am thinking about is the following : you give a stop_point as the starting point and ask for a `bss` fallback.
In this case, I expect the `bss` fallback to be empty (maybe I am wrong here), but it should be tagged as `bss`. 
So here, you remove the `bss` tag and add the `walking` tag. But the fallback has no walking at all.

**Review by kadhikari — COMMENTED** [review]
Is it difficult to find and add a test for this correction ?

File: source/jormungandr/jormungandr/scenarios/journey_filter.py (hunk 3 above)
Comment (xlqian) [reply source/jormungandr/jormungandr/scenarios/journey_filter.py:648]: this case is protected by[ this line](https://github.com/CanalTP/navitia/pull/3478/files/14f6a6e217d8ab86ee93ddf80bbb623030047a99#diff-51842a7f082fe7a1d3dfc36f56a6370e922036194f379d51f0c92c497dc55a69R655)

File: source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py [hunk 4]
```diff
@@ -385,6 +386,14 @@ def _build_crowfly(pt_journey, entry_point, mode, places_free_access, fallback_d
     return _create_crowfly(pt_journey, crowfly_origin, crowfly_destination, begin, end, mode)
 
 
+def _is_pure_walking(dp):
+    if dp.journyes and len(dp.journeys[0].sections) == 1:
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py]:
```suggestion
    if dp.journeys and len(dp.journeys[0].sections) == 1:
```

**Review by pbench — APPROVED** [review]
Looks good to me :+1:

**Comment by xlqian** [comment]
@kadhikari hard to place a tests :( I tried...

**Review by kadhikari — APPROVED** [review]

**Pull request merged**